### DISCUSSION TITLE
Quiz plugin

### DISCUSSION THREAD

**Discussion by msoutopico** [squidfunk/mkdocs-material#5163] [Q&A]
Hi there, 

I am looking for some mkdocs plugin that I could use to add quizzes to my documentation page. Is there such a thing? 

I would like to add a question with a few answer options, only one of which is correct. The user would check the answer they think is correct, and if it 's the correct one, the will see a checkmark, if it's not correct they will see a red cross or something like that. 

If no such plugin exist, perhaps this post should be moved from Q&A to Ideas.
Thanks.

**Comment by squidfunk** [comment]
I don't think there's currently such a plugin. You could consult the [MkDocs Plugins Wiki](https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins) or [create one yourself](https://www.mkdocs.org/dev-guide/plugins/#developing-plugins)!

**Comment by gabrielbdornas** [comment]
For @msoutopico and all interested in this question, I found the [mkdocs-quiz](https://github.com/skyface753/mkdocs-quiz) plugin that does exactly what was asked for, add quizzes to the documentation pages.

[Here](https://fastapidozero.dunossauro.com/quizes/aula_01/) you have a really nice implementation of the plugin. The content of the page is in Portuguese, but I think it's enough to grasp the plugin's potential.

[Here](https://github.com/dunossauro/fastapi-do-zero), the GitHub repo for this project.

It basically:
- [Install](https://github.com/dunossauro/fastapi-do-zero/blob/166ba2e423c1f9e1b4594d951b8f8c8e280159f8/poetry.lock#L1171-L1180) the package[^1].
-  [Adds `mkdocs_quiz` to the `mkdocs.yml` `plugin` property](https://github.com/dunossauro/fastapi-do-zero/blob/166ba2e423c1f9e1b4594d951b8f8c8e280159f8/mkdocs.yml#L82C3-L82C16).
- Adds quizzes based on the [related structure](https://github.com/dunossauro/fastapi-do-zero/blob/166ba2e423c1f9e1b4594d951b8f8c8e280159f8/aulas/quizes/aula_01.md?plain=1#L3-L10):

```markdown
<?quiz?>
question: Are you ready?
answer-correct: Yes!
answer: No!
answer: Maybe!
content:
<h2>Provide some additional content</h2>
<?/quiz?>
```

The project also [better formatted the quiz structure](https://github.com/dunossauro/fastapi-do-zero/blob/main/hooks/quiz_hook.py), if you're interested in taking yours to the next level.

Enjoy!

[^1]: the project uses poetry, but you could simply `pip install mkdocs_quiz` or add `mkdocs_quiz` to your `requirements.txt` file if you're using `venv`.

**Reply by kdkiss** [reply]
I also wrote one if you do a search in GitHub you'll find it let me know if you have questions.

**Reply by jromal** [reply]
> I also wrote one if you do a search in GitHub you'll find it let me know if you have questions.

I tried. But without success. It is not getting as a quiz, but as a regular text.

I like your simple setup for the quiz and therefore I would like to make it work.

**Reply by kdkiss** [reply]
> > I also wrote one if you do a search in GitHub you'll find it let me know if you have questions.
> 
> 
> 
> I tried. But without success. It is not getting as a quiz, but as a regular text.
> 
> 
> 
> I like your simple setup for the quiz and therefore I would like to make it work.

I think it's cause you don't have the css and js files. You may need to copy them manually. Let me know

**Reply by jromal** [reply]
> > > I also wrote one if you do a search in GitHub you'll find it let me know if you have questions.
> > 
> > 
> > I tried. But without success. It is not getting as a quiz, but as a regular text.
> > I like your simple setup for the quiz and therefore I would like to make it work.
> 
> I think it's cause you don't have the css and js files. You may need to copy them manually. Let me know

I cannot see any "css" or "js" referred in your repo. nor on your code.

**Reply by ewels** [reply]
Old topic sorry, but I thought that folks (and future googlers) might be interested that I've given `mkdocs-quiz` a makeover and joined as a maintainer. It now lives here: https://github.com/ewels/mkdocs-quiz

The plugin has had a pretty complete rewrite and does the same thing, but hopefully a bit nicer now and with a bunch of new features (like a progress bar, and a confetti cannon when you finish the quiz).
